chains-examples/docs/whisper-vad/vad_transcribe.py

```diff
@@ -0,0 +1,108 @@
+import asyncio
+import logging
+import time
+from pathlib import Path
+
+import asr_workers.whisper
+import truss_chains as chains
+import utilities.data_types as data_types
+import utilities.helpers as helpers
+import utilities.utilities as utilities
+
+# Configure the logging
+logging.basicConfig(level=logging.INFO)
+
+logger = logging.getLogger(__name__)
+
+
+@chains.mark_entrypoint
+class TranscribeWithVad(chains.ChainletBase):
+    """Transcribes one file end-to-end and sends results to webhook."""
+
+    remote_config = chains.RemoteConfig(
+        docker_image=chains.DockerImage(
+            apt_requirements=["ffmpeg"],
+            # TODO: pin the version of the package
+            pip_requirements=[
+                "pandas",
+                "google-auth",
+                "google-cloud-bigquery",
+                "torch",
+                "torchaudio",
+                "toml",
+                "soundfile==0.10.2",
+            ],
+        ),
+        compute=chains.Compute(predict_concurrency=128),
+    )
+    _context: chains.DeploymentContext
+
+    _whisper: asr_workers.whisper.WhisperModel
+
+    def __init__(
+        self,
+        whisper: asr_workers.whisper.WhisperModel = chains.depends(
+            asr_workers.whisper.WhisperModel, retries=2
+        ),
+        context: chains.DeploymentContext = chains.depends_context(),
+    ) -> None:
+        from chunker.vad import VadOptions
+
+        self._context = context
+        self._whisper = whisper
+        # TODO: expose these parameters as input
+        self._vad_options = VadOptions(
+            threshold=0.5,
+            min_speech_duration_ms=250,
+            max_speech_duration_s=30,
+            min_silence_duration_ms=3000,
+            speech_pad_ms=500,
+        )
+
+    async def run_remote(
+        self,
+        whisper_input: data_types.InputSchema,
```

Comment:
bundling up the whole input like this was done before chains supported optional kwargs. I'd recommend unrolling the input schema so things are less nested and more accessbiel